components/browser/engine-system/src/test/java/mozilla/components/browser/engine/system/SystemEngineSessionTest.kt

```diff
@@ -659,6 +665,22 @@ class SystemEngineSessionTest {
         verify(webView).reload()
     }
 
+    @Test
+    fun desktopModeWithProvidedWideViewPort() {
+        val userAgentMobile = "Mozilla/5.0 (Linux; Android 9) AppleWebKit/537.36 Mobile Safari/537.36"
+        val defaultSettings = DefaultSettings(useWideViewPort = true)
+        val engineSession = spy(SystemEngineSession(testContext, defaultSettings))
+        val webView = mock<WebView>()
+        val webViewSettings = mock<WebSettings>()
+
+        engineSession.webView = webView
+        whenever(webView.settings).thenReturn(webViewSettings)
+        whenever(webViewSettings.userAgentString).thenReturn(userAgentMobile)
+
+        engineSession.toggleDesktopMode(false)
+        verify(webViewSettings).useWideViewPort = true
+    }
```

Comment:
Let's make sure we have a test case for all three cases:
- toggle desktop mode on and off without modifying useWideView (we should have that already?)
- toggle desktop mode on and off with useWideView=true
- toggle desktop mode on and off with useWideView=false